crates/serialize_hierarchy/src/lib.rs

```diff
@@ -63,6 +88,24 @@ where
     }
 }
 
+impl<T> SerializeHierarchy for HashSet<T> {
+    fn serialize_hierarchy(&self, field_path: &str) -> Result<Value> {
+        bail!("cannot access HashSet with path: {}", field_path)
+    }
+
+    fn deserialize_hierarchy(&mut self, field_path: &str, _data: Value) -> Result<()> {
+        bail!("cannot access HashSet with path: {}", field_path)
+    }
+
+    fn exists(_field_path: &str) -> bool {
+        true
+    }
+
+    fn get_hierarchy() -> HierarchyType {
+        HierarchyType::GenericStruct
+    }
+}
+
 impl<T> SerializeHierarchy for Vec<T> {
```

Comment:
Git Blame says, this exists from the beginning. Probably forgotten during implementation. Since there is a task queued for the next iteration to remove the `HierarchyType` and just generate a `BTreeMap` of the contained fields, I would propose to keep it as is.